### ISSUE TITLE
Does Superwall provide "receipt_data" and "transaction_id" ?

### ISSUE THREAD

**Issue by dimpy-iroid** [superwall/Superwall-iOS#274]
I have to send "receipt_data" and "transaction_id" to the server. Where do I get these?

**Comment by anglinb** [comment]
Hey @dimpy-iroid! You can get the transaction id from the SuperwallDelegate 👉  https://superwall.com/docs/3rd-party-analytics#3rd-party-analytics 

Does that seem like what you're looking for?

**Comment by dimpy-iroid** [comment]
Hello @anglinb,
Thank you for the response!

I have implemented it in my code:

```
// MARK: - Superwall Delegate
extension SettingScreen: SuperwallDelegate {
    func handleSuperwallEvent(withInfo eventInfo: SuperwallEventInfo) {
        switch eventInfo.event {
        case .transactionComplete(let transaction, let product, let paywallInfo):
            print("Converted from paywall originalTransactionIdentifier: \(transaction?.originalTransactionIdentifier ?? "")")
            print("Converted from paywall storeTransactionId: \(transaction?.storeTransactionId ?? "")")
            print("Converted from paywall productIdentifier: \(product.productIdentifier)")
            print("Converted from paywall paywallInfo: \(paywallInfo.identifier)")
        case .transactionRestore(let restoreType, let paywallInfo):
            print("transactionRestore restoreType \(restoreType)")
        default:
            print("default \(#function) - \(eventInfo.event)")
        }
    }
}
```

There are two IDs, "originalTransactionIdentifier" and "storeTransactionId"; I'm not sure which ID is the transaction id from these two!!
I tested the subscription on two different devices in the sandbox environment with the same testing account. And I got the same originalTransactionIdentifier in both devices!

**Comment by anglinb** [comment]
The original transaction ID can be thought of like a subscription ID. But it also happens to be the very first transaction in the subscription as well.

**Comment by dimpy-iroid** [comment]
Yes, Thank you for your response. I got the answer. Also, could you please look at this issue? -> https://github.com/superwall/Superwall-iOS/issues/2

**Comment by dimpy-iroid** [comment]
Yes, Thank you for your response. I got the answer.

**Comment by anglinb** [comment]
Can you tell me more about what data you're looking for? We expose a bunch of information on the transaction_complete event: https://superwall.com/docs/tracking-analytics#superwall-events  If there's something missing, we can definitely add it there. We don't currently expose the receipt itself. If you wanted to do verification on your server, would be easier to take the original transaction id and use the App Store Server API instead.

**Comment by anglinb** [comment]
Closing this since I haven't hear back, hopefully that was what you needed!

**Comment by dimpy-iroid** [comment]
Yes, we solved this issue, Thank you for your response!
